From 38c62d9497a2046b961e7471c9d4d013f8425ce8 Mon Sep 17 00:00:00 2001
From: "Le.Ma" <Le.Ma@amd.com>
Date: Tue, 24 Oct 2017 15:31:34 +0800
Subject: [PATCH 2354/4131] drm/amd/display/kcl_dm: Flat core_dc to dc and
 current_context to current_state

The dc_flip_surface_addrs is still used by kcl_dm.

Change-Id: Ica9083e6a4a1d36a2fd9df480ef79ec7748b0c10
Signed-off-by: Le.Ma <Le.Ma@amd.com>
Reviewed-by: Xiaojie Yuan <Xiaojie.Yuan@amd.com>
---
 drivers/gpu/drm/amd/display/dc/core/dc.c | 7 +++----
 1 file changed, 3 insertions(+), 4 deletions(-)

diff --git a/drivers/gpu/drm/amd/display/dc/core/dc.c b/drivers/gpu/drm/amd/display/dc/core/dc.c
index 50379c4..afc8aac 100644
--- a/drivers/gpu/drm/amd/display/dc/core/dc.c
+++ b/drivers/gpu/drm/amd/display/dc/core/dc.c
@@ -1595,7 +1595,6 @@ void dc_flip_plane_addrs(
 		struct dc_flip_addrs flip_addrs[],
 		uint32_t count)
 {
-	struct core_dc *core_dc = DC_TO_CORE(dc);
 	int i, j;
 
 	for (i = 0; i < count; i++) {
@@ -1604,13 +1603,13 @@ void dc_flip_plane_addrs(
 		plane_state->address = flip_addrs[i].address;
 		plane_state->flip_immediate = flip_addrs[i].flip_immediate;
 
-		for (j = 0; j < core_dc->res_pool->pipe_count; j++) {
-			struct pipe_ctx *pipe_ctx = &core_dc->current_context->res_ctx.pipe_ctx[j];
+		for (j = 0; j < dc->res_pool->pipe_count; j++) {
+			struct pipe_ctx *pipe_ctx = &dc->current_state->res_ctx.pipe_ctx[j];
 
 			if (pipe_ctx->plane_state != plane_state)
 				continue;
 
-			core_dc->hwss.update_plane_addr(core_dc, pipe_ctx);
+			dc->hwss.update_plane_addr(dc, pipe_ctx);
 		}
 	}
 }
-- 
2.7.4

